devicetypes/smartthings/zwave-basic-heat-alarm.src/zwave-basic-heat-alarm.groovy

```diff
@@ -0,0 +1,164 @@
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		main "heat"
+		details(["heat", "battery"])
+	}
+}
+
+def installed() {
+	def cmds = []
+	// Device checks in every 4 hours, this interval allows us to miss one check-in notification before marking offline
+	cmds << createEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	cmds << createHeatEvents("heatClear")
+	cmds.each { cmd -> sendEvent(cmd) }
+}
+
+def updated() {
+	// Device checks in every 4 hours, this interval allows us to miss one check-in notification before marking offline
+	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def getCommandClassVersions() {
+	[
+			0x80: 1, // Battery
+			0x84: 1, // Wake Up
+			0x71: 3, // Alarm
+			0x72: 1, // Manufacturer Specific
+	]
+}
+
+def parse(String description) {
+	def results = []
+	if (description.startsWith("Err")) {
+		results << createEvent(descriptionText: description, displayed: true)
+	} else {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			results += zwaveEvent(cmd)
+		}
+	}
+	log.debug "'$description' parsed to ${results.inspect()}"
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	def results = []
+	results << createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+	if (!state.lastbatt || (now() - state.lastbatt) >= 56 * 60 * 60 * 1000) {
+		results << response([
+				zwave.batteryV1.batteryGet().format(),
+				"delay 2000",
+				zwave.wakeUpV1.wakeUpNoMoreInformation().format()
+		])
+	} else {
+		results << response(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	}
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%", isStateChange: true]
+	state.lastbatt = now()
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	return createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def event = [displayed: false]
+	event.linkText = device.label ?: device.name
+	event.descriptionText = "$event.linkText: $cmd"
+	return createEvent(event)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def result = null
+	if (cmd.notificationType == 0x04) {  // Heat Alarm
+		switch (cmd.event) {
+			case 0x00:
+			case 0xFE:
+				result = createHeatEvents("heatClear")
+				break
+			case 0x01: //Overheat detected
+			case 0x02: //Overheat detected Unknown Location
+			case 0x03: //Rapid Temperature Rise
+			case 0x03: //Rapid Temperature Rise Unknown Location
+				result = createHeatEvents("heat")
+				break
+			case 0x07:
+				result = createHeatEvents("tested")
```

Comment:
Just an FYI, the Temperature Alarm capability doesn't define "tested" as a value of the attribute "temperatureAlarm", so you *may* run in to some trouble. (Secret note: The OCF mapping just does a pass-through, so as long as any sort of event marshal doesn't block the event, it should be passed through. If we want a "tested" value, we should look at getting it added.)

Reply:
if it comes up, we can fix it I suppose.

Missed this comment. Sorry.